code/model/model.h

```diff
@@ -346,6 +346,7 @@ class bsp_info
 	bool	use_render_box_offset;		// whether an offset has been defined; needed because one can't tell just by looking at render_box_offset
 	int		use_render_sphere;		// 0==do nothing, 1==only render this object if you are inside the sphere, -1==only if you're outside
 	bool 	use_render_sphere_offset;// whether an offset has been defined; needed because one can't tell just by looking at render_sphere_offset
+	bool    lock_detail_object_distances{false}; // if set should not scale boxes or spheres based on 'model detail' settings
```

Comment:
I would prefer a different term than "lock"; it is not clear without reading the comment.  My first thought was "do_not_scale" but if I understand correctly they *can* scale, just not according to the hard-coded values.

Reply:
Also, how does this PR allow subobjects to not be rendered when they are out of view?

Reply:
Your 1st thought was correct, the new subobject property `$lock_distances` disables scaling of detail box distances - the same distances are used no matter what `model detail` is set to in options.

The code already allows subobjects to be rendered when they are out of view. The high level approach is to set `$detail_sphere` to `-1` (i.e. only render _outside_ the sphere) then use offsets to put the sphere on the opposite side of the ship to the subobject. This all works in current code; EXCEPT when a player sets `model detail` to max (i.e. 5) which multiplies all `$detail_sphere` and `$detail_box` distances by 1.2 - which causes the subobjects to not be drawn on the same side of the ship, instead of only on the other side of the ship. 

So - this PR allows an individual subobject to disable scaling-by-hardcoded-magic-numbers, so the subobjects draw _exactly_ when they are wanted no matter what the player detail settings are.

Anyway, let me know if all that makes sense before I look at possibly changing the variable names.

Reply:
All of that makes sense, except for why multiplying `$detail_sphere` by 1.2 causes it not to work.  Shouldn't that just make it even more negative?

Reply:
The value of `$detail_sphere` isn't being multiplied, rather `$radius` is multiplied by 1.2; which makes the area where the detail_sphere larger; which can result in the sphere (or box) moving to the other side of the ship. And the subobjects don't display when you want them to - on the FotG ISD model it results in really obvious black textures showing when you get close.

For reference; here's an example of a current suboject properties:
```
$detail_sphere: -1
$radius:4000
$offset:0,-4200,200
$lock_distances
```

Yes, we could change the radius such that offset is larger than 4800 (4000*1.2), but that's causing the subobject to display in more situations, the lower you set your graphics options i.e. counter-intuitive to the player

(yes; this way of using `$detail_sphere` is reasonably counter-intuitive as well :))

Reply:
Okay, fair enough.

I'll defer to whatever you think is the more intuitive property name and flag name -- if it's lock_distances instead of do_not_scale_distances, we'll go with that.